cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/CharValidationsPreCompileErrorsTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * (c) Copyright 2016 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.cloudslang.lang.compiler;
+
+import io.cloudslang.lang.compiler.configuration.SlangCompilerSpringConfig;
+import io.cloudslang.lang.compiler.modeller.result.ExecutableModellingResult;
+import java.net.URI;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Bonczidai Levente
+ * @since 8/26/2016
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = SlangCompilerSpringConfig.class)
+public class CharValidationsPreCompileErrorsTest {
```

Comment:
agree, but the tests on top level package (`io.cloudslang.lang.compiler`) are meant to work with real objects - the other sub-packages are tested with mocks